src/main/kotlin/org/rust/ide/annotator/fixes/NameSuggestionFix.kt

```diff
@@ -0,0 +1,45 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.annotator.fixes
+
+import com.intellij.codeInspection.LocalQuickFixOnPsiElement
+import com.intellij.grazie.speller.utils.Distances.levenshtein
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFile
+import org.rust.lang.core.psi.RsPath
+
+/**
+ * Changes the text of some element to the suggested name using the provided function.
+ */
+class NameSuggestionFix(
+    element: PsiElement,
+    private val newName: String,
+    private val replaceElementFn: (name: String) -> PsiElement?
```

Comment:
```suggestion
class NameSuggestionFix<T : PseElement>(
    element: T,
    private val newName: String,
    private val replaceElementFn: (name: String) -> T
```

Let's try to avoid inconsistency between element type and factory method. Of course, you always can use pure `PsiElement` but anyway